src/test/java/com/amazon/dtasdk/utils/HttpUtilsTest.java

```diff
@@ -0,0 +1,41 @@
+package com.amazon.dtasdk.utils;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.net.URI;
+
+public class HttpUtilsTest {
+
+    @Test
+    public void testAppendUri() {
+        Assert.assertEquals("/", HttpUtils.appendUri("", ""));
+        Assert.assertEquals("basepath/two%2Bwords", HttpUtils.appendUri("basepath", "two+words"));
+        Assert.assertEquals("basepath/", HttpUtils.appendUri("basepath", null));
+        Assert.assertEquals("basepath/", HttpUtils.appendUri("basepath/", null));
+        Assert.assertEquals("basepath/test", HttpUtils.appendUri("basepath/", "test"));
+        Assert.assertEquals("basepath/test", HttpUtils.appendUri("basepath/", "/test"));
+        Assert.assertEquals("basepath/test", HttpUtils.appendUri("basepath", "/test"));
+        Assert.assertEquals("basepath/test//doubleslash", HttpUtils.appendUri("basepath", "/test//doubleslash"));
+        Assert.assertEquals("basepath/test/%2Fdoubleslash", HttpUtils.appendUri("basepath", "/test//doubleslash", true));
+    }
+
+    @Test
+    public void testIsUsingNonDefaultPort(){
+        Assert.assertFalse(HttpUtils.isUsingNonDefaultPort(URI.create("http://test:80")));
```

Comment:
Add an assertFalse() for when the port is 0, and for when the port is potentially > 0.

Reply:
https://stackoverflow.com/questions/11225373/negative-port-number-in-http-request